resource-service/common_models/git.go

```diff
@@ -4,68 +4,58 @@ import (
 	"net/url"
 	"strings"
 
+	apimodels "github.com/keptn/go-utils/pkg/api/models"
+
 	kerrors "github.com/keptn/keptn/resource-service/errors"
 )
 
 // GitCredentials contains git credentials info
-type GitCredentials struct {
-	User              string `json:"user,omitempty"`
-	Token             string `json:"token,omitempty"`
-	RemoteURI         string `json:"remoteURI,omitempty"`
-	GitPrivateKey     string `json:"privateKey,omitempty"`
-	GitPrivateKeyPass string `json:"privateKeyPass,omitempty"`
-	GitProxyURL       string `json:"gitProxyUrl,omitempty"`
-	GitProxyScheme    string `json:"gitProxyScheme,omitempty"`
-	GitProxyUser      string `json:"gitProxyUser,omitempty"`
-	GitProxyPassword  string `json:"gitProxyPassword,omitempty"`
-	GitPemCertificate string `json:"gitPemCertificate,omitempty"`
-	// omitempty property is missing due to fallback of this
-	// parameter to "undefined" when marshalling/unmarshalling data
-	// when "false" value is present
-	InsecureSkipTLS bool `json:"insecureSkipTLS"`
-}
+type GitCredentials apimodels.GitAuthCredentials
 
 type GitContext struct {
 	Project     string
 	Credentials *GitCredentials
 }
 
 func (g GitCredentials) Validate() error {
-	if strings.HasPrefix(g.RemoteURI, "https://") || strings.HasPrefix(g.RemoteURI, "http://") {
-		if err := g.validateRemoteURIAndToken(); err != nil {
+	if !strings.HasPrefix(g.RemoteURL, "http://") && !strings.HasPrefix(g.RemoteURL, "ssh://") && !strings.HasPrefix(g.RemoteURL, "https://") {
+		return kerrors.ErrInvalidRemoteURL
+	}
+	if g.HttpsAuth != nil {
+		if err := g.validateRemoteURLAndToken(); err != nil {
 			return err
 		}
 		if err := g.validateProxy(); err != nil {
 			return err
 		}
-	} else if strings.HasPrefix(g.RemoteURI, "ssh://") {
-		if g.GitPrivateKey == "" {
+	} else if g.SshAuth != nil {
+		if g.SshAuth.PrivateKey == "" {
 			return kerrors.ErrCredentialsPrivateKeyMustNotBeEmpty
 		}
 	} else {
-		return kerrors.ErrCredentialsInvalidRemoteURI
+		return kerrors.ErrCredentialsInvalidRemoteURL
```

Comment:
the remote URL should have been already verified at the beginning of the function - should we introduce a specific error type for this case?
Also, let's add a unit test case that reaches this line

Reply:
agree to add a different error, but we have a unit test for this on line 153 in git_test.go